BRCA_Deep_Dive.Rmd

```diff
@@ -0,0 +1,1511 @@
+  
+  POD_supply <- POD_state %>%
+    #dplyr::filter(WRNUM %in% c("61-893", "2061001M00")) %>%
+    dplyr::filter(WRNUM %in% supply_table$water_rights_id) %>%
+    dplyr::distinct(LOCATION, .keep_all = TRUE) %>%
+    dplyr::mutate(
+      name = supply_table %>% 
+        pull (water_system_name)) 
+  
+} else {
+    
+  POD_supply <- POD_state %>%
+    dplyr::filter(
+      OWNER %like% "NATIONAL PARK",
+      str_detect(SUMMARY_ST, "A|P")) %>% # approved / perfected applications
+      #str_detect(USES,"D|M|O|I")) %>%  # filter by POD use category
+    dplyr::distinct(WRNUM,
+                    .keep_all = TRUE) %>%
+    mutate(name = source)
+      
+      #if (nrow(Suppliers) >= 1) {
+      #  POD_supply <- POD_supply %>%
+      #    left_join(., Suppliers, by = WRID)
+      #}
+}
+
+# All PODS within park
+POD_all <- 
+  POD_state %>% 
+  dplyr::filter(park_right == "Park") %>% 
+  dplyr::distinct(WRNUM, .keep_all = TRUE)
+
+
+mapview(POD_state,
+        #zcol = "CFS",
+         layer.name = "State Reported PODs",
+         col.regions = pal[4:6],
+        homebutton = FALSE,
+        cex = 3) +
+mapview(source_table_locs,
+        zcol = "water_system_name",
+        layer.name = "Source:",
+        homebutton = FALSE,
+        col.regions = c("tomato","dodgerblue"),
+        cex = 4) + 
+  mapview(park_boundary,
+          col.regions = "forestgreen",
+          alpha.regions = 0.2,
+          homebutton = FALSE,
+          legend = FALSE) +
+  mapview(source_table_locs$aoi %>% bind_rows(),
+          zcol = "type",
+          layer.name = "AOIs",
+          col.regions = pal,
+          homebutton = FALSE,
+          alpha.regions = 0.2)
+  
+
+
+```
+
+### 5. Nearest NWIS Sites
+
+For many parks, the nearest USGS stream gage is far away. Therefore, here we
+pull in all NWIS gages within 100 km of the park boundary. Of those, we only 
+select stream gages that are considered "reference" gages in the 
+[GAGES-II database (Falcone, 2011)](https://pubs.usgs.gov/publication/70046617).
+We then select the gage that is closest to the park with the most complete
+period of record and delineate associated watershed using the `get_nldi_basin()` 
+function from the {nhdplusTools} package.
+
+```{r, eval = TRUE, message = FALSE, warning = FALSE}
+
+# Get all NWIS sites
+nwis <- listNWIS(aoi = watersupply_watershed %>% 
+                   dplyr::summarize(), dist = .3) 
+
+# Get NWIS Stream Gages
+ref_gages <- get_gagesII(id = nwis$site_no) %>%
+             dplyr::filter(class == "Ref")
+
+nwis_nearest_stream_gage <- nwis %>%
+  dplyr::filter(site_no %in% ref_gages$staid,
+                data_type_cd == "dv") %>%
+  dplyr::left_join(st_drop_geometry(ref_gages), 
+                   by = c("site_no" ="staid")) %>%
+  dplyr::mutate(distance = sf::st_distance(geometry, 
+                                           watersupply_watershed)) %>%
+  dplyr::slice_max(., distance) # grab nearest only
```

Comment:
For nearest distance, should this be `slice_min()` to filter the smallest value?

Reply:
Updated to reflect actual switch from shortest distance to greatest overlapping period of record (1980-2023).